rules/js/react/.esmintrc

```diff
@@ -0,0 +1,74 @@
+{
+  "env": {
+    "browser": true,
+    "es6": true,
+    "mocha": true,
+    "jasmine": true,
+    "node": true,
+    "commonjs": true
+  },
+  "extends": [
+    "airbnb"
+  ],
+  "parserOptions": {
+    "ecmaFeatures": {
+      "jsx": true
+    },
+    "ecmaVersion": 2018,
+    "sourceType": "module"
+  },
+  "globals": {
+    "Atomics": "readonly",
+    "SharedArrayBuffer": "readonly",
+    "afterAll": false,
+    "afterEach": false,
+    "beforeAll": false,
+    "beforeEach": false,
+    "context": false,
+    "describe": false,
+    "expect": false,
+    "it": false,
+    "jest": false,
+    "test": false
+  },
+  "plugins": [
+    "react",
+    "react-hooks",
+    "markdown"
+  ],
+  "rules": {
+    // General
+    // -- Error
+    // -- Warning
+    "class-methods-use-this": 1,
+    "no-else-return": 1,
+    "no-param-reassign": 1,
+    "no-underscore-dangle": 1,
+    "no-unused-vars": 1,
+    "semi": [1, "always"],
+    // -- Off
+    "implicit-arrow-linebreak": 0,
+    "consistent-return": 0,
+    "default-case": 0,
+    "lines-between-class-members": ["off"],
+    "no-shadow": 0,
+    "import/no-unresolved": 0,
+    "import/prefer-default-export": 0,
+    "prefer-destructuring": 0,
```

Comment:
I think we should turn this on, or at the least move it to a warning.

Reply:
Moved no-shadow to error, the rest to warnings, along with some other reevaluations of previously disabled ones.